source/backend/api/Services/AcquisitionFileService.cs

```diff
@@ -425,39 +447,35 @@ public IEnumerable<PimsAgreement> SearchAgreements(AcquisitionReportFilterModel
             return allMatchingAgreements.Where(a => pimsUser.PimsRegionUsers.Any(ur => ur.RegionCode == a.AcquisitionFile.RegionCode));
         }
 
-        public IEnumerable<PimsAgreement> UpdateAgreements(long acquisitionFileId, List<PimsAgreement> agreements)
+        public PimsAgreement UpdateAgreement(long acquisitionFileId, PimsAgreement agreement)
         {
+            _user.ThrowIfNotAuthorized(Permissions.AgreementView);
             _user.ThrowInvalidAccessToAcquisitionFile(_userRepository, _acqFileRepository, acquisitionFileId);
 
-            var currentAcquisitionStatus = GetCurrentAcquisitionStatus(acquisitionFileId);
+            var currentAgreement = _agreementRepository.GetAgreementById(agreement.AgreementId);
 
-            var currentAgreements = _agreementRepository.GetAgreementsByAcquisitionFile(acquisitionFileId);
+            ValidateAcquisitionFileStatus(acquisitionFileId, currentAgreement.AgreementStatusTypeCode);
 
-            var toBeUpdated = currentAgreements.Where(ca => agreements.Any(na => ca.AgreementId == na.AgreementId && !ca.IsEqual(na)));
-            var toBeDeleted = currentAgreements.Where(ca => !agreements.Any(na => ca.AgreementId == na.AgreementId));
+            var updatedAgreement = _agreementRepository.UpdateAgreement(agreement);
+            _agreementRepository.CommitTransaction();
 
-            foreach (var agreement in toBeUpdated)
-            {
-                var agreementStatus = Enum.Parse<AgreementStatusTypes>(agreement.AgreementStatusTypeCode);
-                if (!_statusSolver.CanEditOrDeleteAgreement(currentAcquisitionStatus, agreementStatus) && !_user.HasPermission(Permissions.SystemAdmin))
```

Comment:
is this validation no longer necessary for the agreement?

Reply:
nvm. saw the call to a validate function on 457

Reply:
if so, please just resolve the comment.